### PR TITLE
build: Pre-release publish vscode extension

### LINKED ISSUE #18: Manual Pre-release deployment to vscode marketplace
Vscode extensions can be published / installed as pre-release packages. We should set this up for the vscode-deephaven extension.
https://code.visualstudio.com/api/working-with-extensions/publishing-extension#prerelease-extensions

> Note, that it has been requested that we not deploy this too soon without having some support for Enterprise https://github.com/deephaven/vscode-deephaven/issues/17

- [x] Azure DevOps org - https://dev.azure.com/deephaven-oss/ (will need to request access)
- [x] Visual Studio Publisher - https://marketplace.visualstudio.com/publishers/deephaven

#28  created to setup CI / CD deployments

### PR DESCRIPTION
**Pull request by bmingles**
Extension has been published as a pre-release version to Visual Studio Marketplace
https://marketplace.visualstudio.com/items?itemName=deephaven.vscode-deephaven

We now have the following:
* Azure DevOps org - https://dev.azure.com/deephaven-oss/ (will need to request a user account to be added to the org if you need access)
* Visual Studio Publisher - https://marketplace.visualstudio.com/publishers/deephaven

In this PR:
* Modified .svg logos for targetting 128x128 size without partial pixels
* Generated 128x128 .png logos for the Marketplace logos
* Split out developer specific documentation into CONTRIBUTING.md since README contents gets used as the Marketplace 

resolves #18

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+1. Run the script against the server
+2. Update panels in vscode and deephaven.
+
+On subsequent script runs, the session will be re-used and only steps 4 and 5 will run
+
+### Downloading JS API
+
+The extension dynamically downloads and loads the DH JS API from a running DH Core server.
+
+- `src/jsApi.downloadDhFromServer()`
```
Comment (mofojed) [inline CONTRIBUTING.md:87]: Typo? Confused about this part. Seems like it was carried over from README.md, must have missed it before.

**Review by mofojed — CHANGES_REQUESTED** [review]
Looks good, just a weird bit around the `src/jsapi.downloadDhFromServer()` part. I think that was just copied/pasted, but may as well fix at the same time.

As for publishing - I'm assuming we'll get deephaven-bot to publish from the GitHub action in time with this ticket, and this is just for manually publishing right now: https://github.com/deephaven/vscode-deephaven/issues/28

File: CONTRIBUTING.md (hunk 1 above)
Comment (bmingles) [reply CONTRIBUTING.md]: Fixed.

And yes, #28 will deal with CI deployment presumably from `deephaven-bot`

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -84,5 +84,5 @@ On subsequent script runs, the session will be re-used and only steps 4 and 5 wi
 
 The extension dynamically downloads and loads the DH JS API from a running DH Core server.
 
-- `src/jsApi.downloadDhFromServer()`
-  At runtime, `dh-internal.js` and `dh-core.js` are downloaded from the running DH server (default http://localhost:10000). The files are saved to `out/tmp` as `.cjs` modules, and import / export are converted to cjs compatible ones.
+- `src/dh/dhc.getDhc()`
```
Comment (mofojed) [inline CONTRIBUTING.md:87]: I still find this confusing. Why is the function a bullet point? Why is it calling the function? It seems to break up the paragraph unnecessarily, would just remove the bullet point, have the paragraph go "...running DH Core Server. At runtime ...", then at the end maybe have a link "For implementation details, see [./src/dh/dhc.ts#getDhc](./src/dh/dhc.ts#getDhc)" or something like that.

**Review by mofojed — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 2 above)
Comment (bmingles) [reply CONTRIBUTING.md]: Updated

**Review by mofojed — APPROVED** [review]

**Pull request merged**